### DISCUSSION TITLE
Does Scan support new -retry-tests-on-failure option on Xcode13?

### DISCUSSION THREAD

**Discussion by osrufung** [fastlane/fastlane#19401] [Q&A] [answered]
I've tried, but even the latest iteration succeed, the previous failed iteration count on the global failure counter.

**Comment by kikeenrique** [comment]
Hi,
I'm interested in this _retry_ feature.
Currently scan adds a _number_of_retries_ parameter for retries, but I guess is a hack (it seems not fully reliable given feedback seen in https://github.com/fastlane/fastlane/issues/18587) not backed by recent Xcode additions 

Adding xcodebuild new options for information:
```
-test-iterations <number>
 If specified, tests will run <number> times. May be used in conjunction with either -retry-tests-on-failure or -run-tests-until-failure, in which case this will become the maximum number of iterations.
-retry-tests-on-failure
 If specified, tests will retry on failure. May be used in conjunction with -test-iterations <number>, in which case <number> will be the maximum number of iterations. Otherwise, a maximum of 3 is assumed. May not be used with -run-tests-until-failure.
-run-tests-until-failure
 If specified, tests will run until they fail. May be used in conjunction with -test-iterations <number>, in which case <number> will be the maximum number of iterations. Otherwise, a maximum of 100 is assumed. May not be used with -retry-tests-on-failure.
-test-repetition-relaunch-enabled YES|NO 
 Enable or disable, tests repeating in a new process for each repetition.  Must be used in conjunction with -test-iterations, -retry-tests-on-failure, or -run-tests-until-failure. If not specified, tests will repeat in the same process.
```

Am I right? Is this interesting to add?
Or using xcargs is enought?
```
scan( xcargs: "-test-iterations=3 -retry-tests-on-failure",
          ...
```

**Reply by osrufung** [reply]
I guess the problem is not only with the arguments, but also with the output. It there is any test failed, although a new retry has been solved that issue, scan will interpret that test as failed. 

Each test attempt will be counted as independent tests.

**Reply by kikeenrique** [reply]
Yes, I could reproduced what you say. Using xcargs works as expected with xcodebuild test, but fastlane is not able to detect a successful repeated test.
Anyone does have ideas on this?
Would it be possible to extract info from xcresults file? Is it generated using **result_bundle** param on scan?

**Reply by kikeenrique** [reply]
Some news on this, I've just seen @joshdholtz 💪is working on [integrating Xcode native retry](https://github.com/fastlane/fastlane/commit/187ba0fca009f36af6665e5067f81d7b8e19f48f).

**Reply by osrufung** [reply]
That’s a really good news!! 😁

**Reply by osrufung** [reply]
Seems replace xcpretty with trainer will be the first step 😍 

https://github.com/fastlane/fastlane/pull/19624

**Comment by osrufung** [comment, marked as answer]
And finally here is the answer to that :) 
https://github.com/fastlane/fastlane/pull/19629
